domain/application/service/migration.go

```diff
@@ -295,19 +296,45 @@ func (s *MigrationService) ImportApplication(ctx context.Context, name string, a
 		return errors.Annotatef(err, "creating application args")
 	}
 
-	units := args.Units
-	numUnits := len(units)
-	appArg.Scale = numUnits
+	appArg.Scale = len(args.Units)
+	unitArgs, err := makeUnitArgs(args.Units)
+	if err != nil {
+		return errors.Annotatef(err, "creating unit args")
+	}
+
+	appID, err := s.st.CreateApplication(ctx, name, appArg, nil)
+	if err != nil {
+		return errors.Annotatef(err, "creating application %q", name)
+	}
+	if modelType == model.IAAS {
+		err = s.st.InsertMigratingIAASUnits(ctx, appID, unitArgs...)
+	} else {
+		err = s.st.InsertMigratingCAASUnits(ctx, appID, unitArgs...)
+	}
```

Comment:
If we want to push the units into the new `ImportApplication` method as *one* transaction, we're then likely to want to have `ImportIAASApplication` and `ImportCAASApplication`? That way we can drop the `InsertMigrating...` public methods to package level.

Reply:
Yes. I suggested in the PR description we'd want to do this but felt it would be in a follow up PR.